#### [시작하기]

<div class="alert alert-info">
    <font color=black>
    Pandas를 사용하기 위해서 처음 할 일을 무엇인가요?
    </font>
</div>

In [1]:
import pandas as pd # 판다스를 pd로 축약하여 불러옵니다

# **[7] Pandas로 데이터 확인하기**

[오늘 배울 것]
1. 데이터를 출력해 확인할 수 있다.
2. 데이터에 대한 정보를 확인할 수 있다.


##  **1. Pandas로 데이터 출력**
### **(1) 데이터 불러오기**

> 우선 데이터가 있어야 그 내용을 확인할 수 있습니다.  
이번에도 seaborn 패키지에서 만들어진 데이터를 불러 오도록 하겠습니다. 
>
> `print(Sereis)` 또는 `print(DataFrame)` 으로 데이터를 출력해봅시다.

In [3]:
# 예제 1: 제공되는 데이터 불러와서 출력해보기

import seaborn as sns
new_df = sns.load_dataset('planets')

print(new_df)

               method  number  orbital_period   mass  distance  year
0     Radial Velocity       1      269.300000   7.10     77.40  2006
1     Radial Velocity       1      874.774000   2.21     56.95  2008
2     Radial Velocity       1      763.000000   2.60     19.84  2011
3     Radial Velocity       1      326.030000  19.40    110.62  2007
4     Radial Velocity       1      516.220000  10.50    119.47  2009
...               ...     ...             ...    ...       ...   ...
1030          Transit       1        3.941507    NaN    172.00  2006
1031          Transit       1        2.615864    NaN    148.00  2007
1032          Transit       1        3.191524    NaN    174.00  2007
1033          Transit       1        4.125083    NaN    293.00  2008
1034          Transit       1        4.187757    NaN    260.00  2008

[1035 rows x 6 columns]


> `print` 함수는 데이터가 너무 큰 경우에 위와 같이 중간을 생략해서 출력합니다.  

> `print`를 이용해 출력한 것으로 알 수 있는 것은 무엇인가요?  
>-  method 열에는 Radial Velocity, Transit이란 값이 있다.  
>- number 열에는 1만 들어있다.  
>- mass 열에는 데이터 값이 없는 경우(NaN)가 있다.

### **(2) 데이터 원하는 만큼 출력하기**




> Pandas 데이터를 출력하는 또다른 방법에는 `head`, `tail`이 있습니다.  
> 이 방법들의 특징은 원하는 길이만큼을 출력할 수 있다는 점입니다. 
</br>  
> `Series.head(n)`, `DataFrame.head(n)`: 위에서부터 n개의 행을 출력  
> `Series.tail(n)`, `DataFrame.tail(n)`: 아래에서부터 n개의 행을 출력

In [8]:
# 예제 2: 데이터의 앞순서부터 원하는 만큼 출력하기

print("데이터 앞에서부터 20개 출력:")
new_df.head(20)

데이터 앞에서부터 20개 출력:


method  number  orbital_period    mass  distance  year
0   Radial Velocity       1      269.300000   7.100     77.40  2006
1   Radial Velocity       1      874.774000   2.210     56.95  2008
2   Radial Velocity       1      763.000000   2.600     19.84  2011
3   Radial Velocity       1      326.030000  19.400    110.62  2007
4   Radial Velocity       1      516.220000  10.500    119.47  2009
5   Radial Velocity       1      185.840000   4.800     76.39  2008
6   Radial Velocity       1     1773.400000   4.640     18.15  2002
7   Radial Velocity       1      798.500000     NaN     21.41  1996
8   Radial Velocity       1      993.300000  10.300     73.10  2008
9   Radial Velocity       2      452.800000   1.990     74.79  2010
10  Radial Velocity       2      883.000000   0.860     74.79  2010
11  Radial Velocity       1      335.100000   9.880     39.43  2009
12  Radial Velocity       1      479.100000   3.880     97.28  2008
13  Radial Velocity       3     1078.000000   2.530     14.08  1996
14  Radial Velocity       3     2391.000000   0.540     14.08  2001
15  Radial Velocity       3    14002.000000   1.640     14.08  2009
16  Radial Velocity       1        4.230785   0.472     15.36  1995
17  Radial Velocity       5       14.651000   0.800     12.53  1996
18  Radial Velocity       5       44.380000   0.165     12.53  2004
19  Radial Velocity       5     4909.000000   3.530     12.53  2002

> `head`를 이용해 출력한 것으로 더 알 수 있는 것은 무엇인가요?  
>- number 열에는 1보다 큰 값도 있다.   

In [9]:
# 예제 3: 데이터의 뒷순서부터 원하는 만큼 출력하기

print("데이터 뒤에서부터 20개 출력:")
new_df.tail(20)

데이터 뒤에서부터 20개 출력:


method  number  orbital_period  mass  distance  year
1015  Transit       1        1.573292   NaN     350.0  2012
1016  Transit       1        2.311424   NaN     310.0  2013
1017  Transit       1        4.086052   NaN     380.0  2012
1018  Transit       1        4.614420   NaN     225.0  2012
1019  Transit       1        2.903675   NaN     345.0  2012
1020  Transit       1        2.216742   NaN     340.0  2012
1021  Transit       1        2.484193   NaN     260.0  2013
1022  Transit       1        1.360031   NaN      93.0  2012
1023  Transit       1        2.175176   NaN     550.0  2012
1024  Transit       1        3.662387   NaN     240.0  2012
1025  Transit       1        3.067850   NaN      60.0  2012
1026  Transit       1        0.925542   NaN     470.0  2014
1027  Imaging       1             NaN   NaN      19.2  2011
1028  Transit       1        3.352057   NaN    3200.0  2012
1029  Imaging       1             NaN   NaN      10.1  2012
1030  Transit       1        3.941507   NaN     172.0  2006
1031  Transit       1        2.615864   NaN     148.0  2007
1032  Transit       1        3.191524   NaN     174.0  2007
1033  Transit       1        4.125083   NaN     293.0  2008
1034  Transit       1        4.187757   NaN     260.0  2008

> `tail`을 이용해 출력한 것으로 더 알 수 있는 것은 무엇인가요?  
>- method 열에는 Imaging 이란 값도 있다.  

#### <연습문제>
<a class="jcecodle-mk jcecodle-mk__problem-chip" data-problem-id="3urbitKPCt8" href='#jce-judge/open/3urbitKPCt8' target='_self'><span>판다스 기초 - 데이터 출력하기</span></a>




## **2. Pandas로 데이터 정보 확인**

> 데이터의 일부분을 출력하는 것으로는 전체적인 데이터를 파악하기에 무리가 있습니다.  
> 데이터가 크다면 전체를 출력하여도 마찬가지입니다.  
>
> Pandas에는 데이터에 대한 전반적인 정보를 정리해주는 기능들이 있습니다.  

### (1) 데이터 기본 정보 확인하기
> `Series.info()`를 통해 다음을 알 수 있습니다.   
>- 총 행의 길이
>- Series의 이름
>- 데이터 갯수 (NaN 제외)
>- 데이터 타입
>- 데이터 용량

> `DataFrame.info()`를 통해 다음을 알 수 있습니다.   
>- 총 행의 길이
>- 각 열의 이름
>- 각 열에 존재하는 데이터 갯수 (NaN 제외)
>- 각 열의 데이터 타입
>- 데이터 용량

In [15]:
# 예제 4: 데이터의 기본 정보 확인하기

new_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1035 entries, 0 to 1034
Data columns (total 6 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   method          1035 non-null   object 
 1   number          1035 non-null   int64  
 2   orbital_period  992 non-null    float64
 3   mass            513 non-null    float64
 4   distance        808 non-null    float64
 5   year            1035 non-null   int64  
dtypes: float64(3), int64(2), object(1)
memory usage: 48.6+ KB


> `info`를 이용해 알아낸 정보를 정리해봅시다.
>- 총 행의 길이: 1035
>- method 열에는 1035개의 object타입 데이터가 있다.
>- number 열에는 1035개의 int타입 데이터가 있다.
>- orbital_period 열에는 992개의 float타입 데이터가 있다.
>- mass 열에는 513개의 float타입 데이터가 있다.
>- distance 열에는 808개의 float타입 데이터가 있다.
>- year 열에는 1035개의 int타입 데이터가 있다.
>- 데이터 용량: 약 48.6 KB 

#### <연습문제>
<a class="jcecodle-mk jcecodle-mk__problem-chip" data-problem-id="HA4xV3SU1_A" href='#jce-judge/open/HA4xV3SU1_A' target='_self'><span>판다스 기초 - 데이터 정보 확인하기 (1)</span></a>

<a class="jcecodle-mk jcecodle-mk__problem-chip" data-problem-id="SeZBiTED_n4" href='#jce-judge/open/SeZBiTED_n4' target='_self'><span>판다스 기초 - 데이터 정보 확인하기 (2)</span></a>







### (2) 데이터 통계 정보 확인하기
> `Sereis.describe()`, `DataFrame.describe()`를 통해 다음을 알 수 있습니다.   
>- count: 데이터 갯수 (NaN 제외)
>- mean: 평균값
>- std: 표준편차
>- min: 최솟값
>- 25%: 크기 순으로 정렬했을때 4분의 1번째에 해당하는 값
>- 50%: 크기 순으로 정렬했을때 중앙에 해당하는 값 (중앙값)
>- 75%: 크기 순으로 정렬했을때 4분의 3번째에 해당하는 값
>- max: 최댓값
>
> 위의 통계량은 숫자에 대한 것으로 데이터의 값이 어떻게 분포되어 있는지 알려줍니다.

In [18]:
# 예제 5: 숫자 데이터의 통계 정보 확인하기

new_df.describe()

number  orbital_period        mass     distance         year
count  1035.000000      992.000000  513.000000   808.000000  1035.000000
mean      1.785507     2002.917596    2.638161   264.069282  2009.070531
std       1.240976    26014.728304    3.818617   733.116493     3.972567
min       1.000000        0.090706    0.003600     1.350000  1989.000000
25%       1.000000        5.442540    0.229000    32.560000  2007.000000
50%       1.000000       39.979500    1.260000    55.250000  2010.000000
75%       2.000000      526.005000    3.040000   178.500000  2012.000000
max       7.000000   730000.000000   25.000000  8500.000000  2014.000000

> `describe`를 통해 새롭게 알아낸 정보를 정리해봅시다.  
> - number 열에는 1부터 7까지의 값이 있다.
> - number 열의 최솟값, 중앙값이 1 이므로 절반 이상의 데이터가 1이다.
> - year 열에는 2007부터 2014까지의 값이 있다. 

#### <연습문제>
<a class="jcecodle-mk jcecodle-mk__problem-chip" data-problem-id="T0QwboxCL-A" href='#jce-judge/open/T0QwboxCL-A' target='_self'><span>판다스 기초 - 숫자 데이터 통계 확인하기</span></a>


> 위의 결과에서 method 열의 데이터는 숫자가 아니기 때문에 통계량이 출력되지 않았습니다.   
> `PandasData.describe(include='O')` 를 이용하면 object 타입 데이터에 대한 다음 통계량을 알려줍니다.  
>- count: 데이터 갯수 (NaN 제외)
>- unique: 데이터의 고유값 종류 갯수
>- top: 가장 수가 많은 고유값
>- freq: top에 해당하는 고유값의 갯수

In [19]:
# 예제 6: Object 타입 데이터의 통계 정보 확인하기

new_df.describe(include='O')

method
count              1035
unique               10
top     Radial Velocity
freq                553

> 출력하기를 했을때는 method 열의 데이터를 총 3가지 밖에 확인하지 못했습니다.    
> `describe`를 이용하면 전체적으로 몇가지의 고유값이 있는지 쉽게 알 수 있습니다.  
>
> 새로 알아낸 정보를 정리해봅시다.
>- method 열에는 총 10가지 종류의 값이 있다.
>- Radial Velocity가 가장 많고 1035개 데이터 중 553개를 차지한다.

#### <연습문제>
<a class="jcecodle-mk jcecodle-mk__problem-chip" data-problem-id="c1xby3pplU0" href='#jce-judge/open/c1xby3pplU0' target='_self'><span>판다스 기초 - 문자 데이터 통계 확인하기</span></a>





